### PR TITLE
make a library out of control app crate

### PR DESCRIPTION
**Pull request by Beaudidly**
This lets me use your control work as a library! Important for creating the application wrapper around the control mngr.

(Also some small fixes because I had some errors on compiling regarding your logging macro calls)

### CODE REVIEW

File: control_apps/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+mod valve;
+
+pub mod control_mngr;
```
Comment (philiplinden) [inline control_apps/src/lib.rs]: add newline at end of file

File: control_apps/src/control_mngr.rs [hunk 2]
```diff
@@ -8,6 +8,8 @@ use std::fmt;
 use std::thread::sleep; // only used to emulate some sort of POST
 use std::time::Duration;
 
+use log::{info, warn, debug};
```
Comment (philiplinden) [inline control_apps/src/control_mngr.rs:11]: no error log? is that because there are no error statements in this crate?

**Review by philiplinden — COMMENTED** [review]
i see, so the `lib.rs` makes the `control_mgr` crate public but the other supporting crate(s) like `valve` are not public if this app is used elsewhere.

makes sense. how do I demo that these changes actually achieve that?

**Review by philiplinden — APPROVED** [review]

File: control_apps/src/control_mngr.rs (hunk 2 above)
Comment (Beaudidly) [reply control_apps/src/control_mngr.rs:11]: yes

File: control_apps/src/lib.rs (hunk 1 above)
Comment (Beaudidly) [reply control_apps/src/lib.rs]: done

**Pull request merged**